OneDriveApiBrowser/FormBrowser.cs

```diff
@@ -271,71 +281,87 @@ private void UpdateConnectedStateUx(bool connected)
 
         private async Task SignIn(ClientType clientType)
         {
-            if (this.oneDriveClient == null)
+            if (clientType == ClientType.Consumer)
```

Comment:
I see a bunch of code that I thing can be trimed down for clarty, instead of forking completely for MSA vs AAD instead fork just to create the authenication task that has to be awaited, then have a single check for an auth failure flow and create/set the OneDriveClient object